app/scripts/components/exploration/components/timeline/timeline-controls.tsx

```diff
@@ -87,8 +90,132 @@ export function TimelineDateAxis(props: Omit<TimelineControlsProps, "onZoom">) {
   );
 }
 
+export const TIMELINE_PLAYHEAD_COLOR_PRIMARY = '#8b8b8b';
+export const TIMELINE_PLAYHEAD_COLOR_SECONDARY = '#333333';
+export const TIMELINE_PLAYHEAD_COLOR_TEXT = '#ffffff';
+export const TIMELINE_PLAYHEAD_COLOR_LABEL = '#cccccc';
+
+const TimelineHeadIndicatorsWrapper = styled.div`
+  position: absolute;
+  bottom: -30px;
+  width: 100%;
+`;
+
+const TimelinePlayheadBase = styled.div`
+  background-color: ${TIMELINE_PLAYHEAD_COLOR_PRIMARY};
+  color: ${TIMELINE_PLAYHEAD_COLOR_TEXT};
+  padding: 3px;
+  border-radius: 4px;
+  font-size: 0.75rem;
+  position: relative;
+  width: max-content;
+  font-weight: 500;
+
+  &::after, &::before {
+    content: '';
+    position: absolute;
+    bottom: 1px;
+    width: 0;
+    height: 0;
+    border-top: 12px solid transparent;
+    border-bottom: 12px solid transparent;
+  }
+`;
+
+const TimelinePlayheadLeftIndicator = styled(TimelinePlayheadBase)`
+  background-color: ${TIMELINE_PLAYHEAD_COLOR_SECONDARY};
+
+  &::after {
+    left: -6%;
+    border-right: 7px solid ${TIMELINE_PLAYHEAD_COLOR_SECONDARY};
+  }
+`;
+
+const TimelinePlayheadLeftIndicatorSecondary = styled(TimelinePlayheadBase)`
+  &::after {
+    left: -30%;
+    border-right: 7px solid ${TIMELINE_PLAYHEAD_COLOR_PRIMARY};
+  }
+`;
+
+const TimelinePlayheadRightIndicator = styled(TimelinePlayheadBase)`
+  background-color: ${TIMELINE_PLAYHEAD_COLOR_SECONDARY};
+
+  &::before {
+    right: -6%;
+    border-left: 7px solid ${TIMELINE_PLAYHEAD_COLOR_SECONDARY};
+  }
+`;
+
+const TimelinePlayheadRightIndicatorSecondary = styled(TimelinePlayheadBase)`
+  &::before {
+    right: -30%;
+    border-left: 7px solid ${TIMELINE_PLAYHEAD_COLOR_PRIMARY};
+  }
+`;
+
+const TimelineHeadRightIndicators = styled.div`
+  position: absolute;
+  bottom: 0;
+  right: 120px;
+  display: flex;
+  flex-direction: row-reverse;
+  gap: 10px;
+`;
+
+
+const TimelineHeadLeftIndicators = styled.div`
+  position: absolute;
+  bottom: 0;
+  left: -8px;
+  display: flex;
+  gap: 10px;
+`;
+
+const getIndicators = (outOfViewHeads: TimelineHead[]) => {
```

Comment:
🤔 Is there a reason you are creating this like a hook fn instead of a functional component? its returning a JSX.Element and can accept `outOfViewHeads` as a prop. If we do it like this doesn't it make sense to place this inside the functional component and wrap it with a useCallback with `outOfViewHeads` as dep to prevent unnecessary recreation? I think with how its done right now, its going to be unnessarily called? correct me if i'm wrong

Reply:
Aside from the lowerCase naming, this is already a functional component:P But you're completely right we should make sure it doesn't re-render, so I re-factored it to make use of [memo](https://react.dev/reference/react/memo). Let me know what do you think.

Reply:
hm technically it is not a functional component. Its just a function that returns JSX - `getIndicators`. Difference is functional components are something called like `<Indictors {...props} />` while function that returns JSX is called like `<>{getIndictors()}</>`